flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenUtils.scala

```diff
@@ -203,6 +210,569 @@ object CodeGenUtils {
       throw new CodeGenException("Integer expression type expected.")
     }
 
+  def generateNullLiteral(
```

Comment:
Actually these methods should be part of the `ExprCodeGenerator`. They are more than just utils but the actual main logic.